adapters/zentotem/zentotem.go

```diff
@@ -0,0 +1,131 @@
+package zentotem
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+	"net/http"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the {bidder} adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	if request == nil {
+		return nil, []error{&errortypes.BadInput{Message: "empty request"}}
+	}
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{Message: "empty request.Imp"}}
+	}
+
+	requests := make([]*adapters.RequestData, 0, len(request.Imp))
+	var errors []error
+
+	requestCopy := *request
+	for _, imp := range request.Imp {
+		requestCopy.Imp = []openrtb2.Imp{imp}
+
+		requestJSON, err := json.Marshal(request)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		requestData := &adapters.RequestData{
+			Method: "POST",
+			Uri:    a.endpoint,
+			Body:   requestJSON,
+			ImpIDs: []string{imp.ID},
+		}
+		requests = append(requests, requestData)
+	}
+	return requests, errors
+}
+
+func getMediaTypeForBid(imp openrtb2.Imp) (openrtb_ext.BidType, error) {
```

Comment:
Requesting you to please re-write this function to use MType field.

Reply:
fixed